### PR TITLE
Fix Safari docs crash

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/5068

I was able to debug with my local build

We were getting into a state where the offsetWidth was within 1px of the scrollWidth, resulting in trying to make the font ever smaller since there was no guard against it getting too small/negative

The reason we were within 1px is that Safari seems to `floor` clientWidth when other browsers do `round`, so I've added a + 1 to account for that. I tried using the bounding client rect and rounding myself, but for some reason, it sometime suffer from being a pixel off as well. So it may be a deeper rounding issue than I can account for.

the + 1 discrepancy doesn't appear to harm anything and it was taking wayyyy too much of my time to try to fix it.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

File: packages/dev/docs/src/client.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ if (process.env.NODE_ENV === 'production') {
   window.addEventListener('load', () => listen());
 }
 
+let raf = null;
```
Comment (snowystinger) [inline packages/dev/docs/src/client.js:27]: this is just some defensive code

**Comment by theMosaad** [comment]
Nice one 🙌  Can confirm it fixed the issue for me

**Review by reidbarber — APPROVED** [review]
Fixes the issue when zooming. I'll have to re-test this after release to see if it fixes the crashing I experience in the production build without zooming.

**Review by LFDanLu — APPROVED** [review]
verified that I could now reproduce the previous issue on prod docs in Safari and that the changes here have fixed the crash

**Pull request merged**